cirq/ops/common_channels.py

```diff
@@ -50,90 +60,156 @@ def __init__(self, p_x: float, p_y: float, p_z: float) -> None:
+            Pi = np.identity(1)
+            for gate in pauli:
+                if gate == 'I':
+                    Pi = np.kron(Pi, protocols.unitary(identity.I))
+                elif gate == 'X':
+                    Pi = np.kron(Pi, protocols.unitary(pauli_gates.X))
+                elif gate == 'Y':
+                    Pi = np.kron(Pi, protocols.unitary(pauli_gates.Y))
+                elif gate == 'Z':
+                    Pi = np.kron(Pi, protocols.unitary(pauli_gates.Z))
+            Pis.append(Pi)
+        return tuple(zip(self._pis.values(), Pis))
+
+    def _num_qubits_(self) -> int:
+        return self._num_qubits
 
     def _has_mixture_(self) -> bool:
         return True
 
     def _value_equality_values_(self):
-        return self._p_x, self._p_y, self._p_z
+        return self._num_qubits, hash(tuple(sorted(self._pis.items())))
 
     def __repr__(self) -> str:
-        return 'cirq.asymmetric_depolarize(p_x={!r},p_y={!r},p_z={!r})'.format(
-            self._p_x, self._p_y, self._p_z
-        )
+        return (
+            'cirq.asymmetric_depolarize' +
+            '(num_qubits={!r},pis={!r})'.format(self._num_qubits, self._pis))
 
     def __str__(self) -> str:
-        return 'asymmetric_depolarize(p_x={!r},p_y={!r},p_z={!r})'.format(
-            self._p_x, self._p_y, self._p_z
-        )
+        return 'asymmetric_depolarize(num_qubits={!r},pis={!r})'.format(
+            self._num_qubits, self._pis)
 
     def _circuit_diagram_info_(self,
                                args: 'protocols.CircuitDiagramInfoArgs') -> str:
-        if args.precision is not None:
-            f = '{:.' + str(args.precision) + 'g}'
-            return 'A({},{},{})'.format(f, f, f).format(self._p_x, self._p_y,
-                                                        self._p_z)
-        return 'A({!r},{!r},{!r})'.format(self._p_x, self._p_y, self._p_z)
+        if self._num_qubits == 1:
+            if args.precision is not None:
+                f = '{:.' + str(args.precision) + 'g}'
+                return 'A({},{},{})'.format(f, f,
+                                            f).format(self.p_x, self.p_y,
+                                                      self.p_z)
+            return 'A({!r},{!r},{!r})'.format(self.p_x, self.p_y, self.p_z)
+        else:
+            if args.precision is not None:
+                f = '{:.' + str(args.precision) + 'g}'
+                ps = [
+                    pauli + ':' + '{}'.format(f).format(p)
+                    for pauli, p in self._pis.items()
+                ]
+            else:
+                ps = [
+                    pauli + ':' + '{!r}'.format(p)
+                    for pauli, p in self._pis.items()
+                ]
+            return 'A({})'.format(', '.join(ps))
 
     @property
     def p_x(self) -> float:
         """The probability that a Pauli X and no other gate occurs."""
-        return self._p_x
+        assert self._num_qubits == 1
```

Comment:
This raises `AssertionError` which should be raised for conditions which we believe should not arise unless there is a bug.

We should raise `ValueError` which signifies that the user of this class made a mistake.

(Applies below, too.)

Reply:
Done.